### PR TITLE
fix(atomic): make atomic-result-link default/unnamed slot render properly

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1643

discuss: https://discuss.coveo.com/t/support-00080206-atomic-result-link-is-not-generating-links-with-display-type-list-post-v1-46-0/7873

### CODE REVIEW

File: packages/atomic/src/utils/slot-utils.ts [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+export function getDefaultSlotFromHost(host: HTMLElement) {
+  const children = Array.from(host.children);
+  const defaultSlot = children.filter((child) => !child.hasAttribute('slot'));
+
+  if (defaultSlot.length > 1) {
+    console.warn('Element should only have 1 default slot.', host);
   }
 
-  return slots[0];
+  return defaultSlot[0];
```
Comment (olamothe) [inline packages/atomic/src/utils/slot-utils.ts:26]: Is it possible that `defaultSlot` would somehow be an empty array ? And if so should there be a check for this ?

File: packages/atomic/src/utils/slot-utils.ts [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
+  if (targetSlot.length > 1) {
+    console.warn(`Element should only have 1 slot named "${slotName}".`, host);
+  }
+
+  return targetSlot[0];
+}
+
+export function getDefaultSlotFromHost(host: HTMLElement) {
+  const children = Array.from(host.children);
+  const defaultSlot = children.filter((child) => !child.hasAttribute('slot'));
```
Comment (btaillon) [inline packages/atomic/src/utils/slot-utils.ts:20]: it looks like prior to these changes, it was possible to have both a slot and a "default" slot by giving an element `slot=""`. Was this the case? That no longer looks possible here.

Comment (ThibodeauJF) [reply packages/atomic/src/utils/slot-utils.ts:20]:
```
<atomic-result-link>
  <h1>hello</h1>
</atomic-result-link>

<atomic-result-link>
  <h1 part=''>hello</h1>
</atomic-result-link>

<atomic-result-link>
  <h1>hello</h1>
  <a part="attributes" target="_blank" title="HELLO"></a>
</atomic-result-link>

<atomic-result-link>
  <h1 part=''>hello</h1>
  <a part="attributes" target="_blank" title="HELLO"></a>
</atomic-result-link>
```

Before:
Screen Shot 2022-05-19 at 2 21 23 PM
Screen Shot 2022-05-19 at 2 21 23 PM

After:
Screen Shot 2022-05-19 at 2 20 33 PM
Screen Shot 2022-05-19 at 2 20 33 PM

Comment (ThibodeauJF) [reply packages/atomic/src/utils/slot-utils.ts:20]:
```
<atomic-result-link>
  <h1>hello</h1>
</atomic-result-link>

<atomic-result-link>
  <h1 part=''>hello</h1>
</atomic-result-link>

<atomic-result-link>
  <h1>hello</h1>
  <a part="attributes" target="_blank" title="HELLO"></a>
</atomic-result-link>

<atomic-result-link>
  <h1 part=''>hello</h1>
  <a part="attributes" target="_blank" title="HELLO"></a>
</atomic-result-link>
```

Before:
Screen Shot 2022-05-19 at 2 21 23 PM
Screen Shot 2022-05-19 at 2 21 23 PM

After:
Screen Shot 2022-05-19 at 2 20 33 PM
Screen Shot 2022-05-19 at 2 20 33 PM

Comment (btaillon) [reply packages/atomic/src/utils/slot-utils.ts:20]: In the example you gave me, you're using `part`, did you test it using `part` or `slot`?

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/utils/slot-utils.ts (hunk 2 above)
Comment (ThibodeauJF) [reply packages/atomic/src/utils/slot-utils.ts:20]: ARGH always confuse those 2

Comment (ThibodeauJF) [reply packages/atomic/src/utils/slot-utils.ts:20]: fixed!

**Pull request merged**